specs/chain/hs/src/Cardano/Spec/Chain/STS/Block.hs

```diff
@@ -83,11 +83,8 @@ hashHeader = hashlazy . pack . show
 
 
 -- | Compute the epoch for the given _absolute_ slot
-sEpoch :: Slot -> Epoch
-sEpoch (Slot s) = Epoch $ s `div` slotsPerEpoch
-  where
-    -- Hardcoded number of slots per epoch, as per Byron.
-    slotsPerEpoch = 21600
+sEpoch :: Slot -> SlotCount -> Epoch
```

Comment:
TBH I think the previous function was better, since at least that could encapsulate the "pv to slots-per-epoch" map which we will ultimately want. This would only work for a fixed-but-configurable slots per epoch, whereas we know we will have non-fixed non-configurable values.

Reply:
Problem is that in the actual implementation the number of slots per epoch depends on `k`. So having this hardcoded constant here will break the tests. 

I forgot when do we want to make the slots per-epoch configurable. If it is in Shelley I guess we don't have to worry about it here. If it is during the handover phase then we might need to rethink this.

But all in all, even if we have a `pv to slots per epoch` it seems that this function would have to be changed to take the protocol version as additional parameter.